collector/msft_fsrmquota.go

```diff
@@ -0,0 +1,232 @@
+package collector
+
+import (
+	//"strings"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+func init() {
+	Factories["msft_fsrmquota"] = NewMSFT_FSRMQuotaCollector
+}
+
+// A MSFT_FSRMQuotaCollector is a Prometheus collector for WMI MSFT_FSRMQuota metrics
+type MSFT_FSRMQuotaCollector struct {
+	QuotasCount *prometheus.Desc
+	Path      *prometheus.Desc
+	PeakUsage *prometheus.Desc
+	Size      *prometheus.Desc
+	Usage     *prometheus.Desc
+	
+	
+	Description           *prometheus.Desc
+	Disabled              *prometheus.Desc
+	MatchesTemplate       *prometheus.Desc
+	SoftLimit             *prometheus.Desc
+	Template              *prometheus.Desc
+	Threshold             *prometheus.Desc
+	
+}
+
+// NewMSFT_FSRMQuotaCollector ...
+func NewMSFT_FSRMQuotaCollector() (Collector, error) {
+	const subsystem = "msft_fsrmquota"
+	return &MSFT_FSRMQuotaCollector{
+		QuotasCount: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "count"),
+			"Number of Quotas",
+			nil,
+			nil,
+		),
+		PeakUsage: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "peak_usage"),
+			"The highest amount of disk space usage charged to this quota. (PeakUsage)",
+			[]string{"quotaPath","quotaTemplate"},
+			nil,
+		),
+		Path: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "path"),
+			"A string that represents a valid local path to a folder. Must not exceed the value of MAX_PATH. Required. (Path)",
+			[]string{"quotaPath","quotaTemplate"},
+			nil,
+		),
+		Size: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "size"),
+			"The size of the quota. If the Template property is not provided then the Size property must be provided (Size)",
+			[]string{"quotaPath","quotaTemplate"},
+			nil,
+		),
+		Usage: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "usage"),
+			"The current amount of disk space usage charged to this quota. (Usage)",
+			[]string{"quotaPath","quotaTemplate"},
+			nil,
+		),
+		Description: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "description"),
+			"A string up to 1KB in size. Optional. The default value is an empty string. (Description)",
```

Comment:
The help from the docs doesn't seem very helpful, so maybe just something like this?
```suggestion
			"Description of the quota (Description)",
```
I'm a bit worried about the "up to 1KB in size", though. It would probably break a few assumptions in visualisation tools, for example. Perhaps we should truncate the length of the label to something shorter? This might be unexpected to the user, on the other hand...